src/lucky_migrator/create_table_statement.cr

```diff
@@ -1,65 +1,48 @@
 class LuckyMigrator::CreateTableStatement
-  getter statements = [] of String
-  getter rows = [] of String
-  getter table_statement = ""
-  getter index_statements = [] of String
+  private getter rows = [] of String
+  private getter index_statements = [] of String
 
   def initialize(@table_name : Symbol)
   end
 
   # Accepts a block to build a table and indices using `add` and `add_index` methods.
   #
-  # The generated sql statements are aggregated in the statements getter, and the
-  # table_statement and index_statements are available individually as getters.
+  # The generated sql statements are aggregated in the `statements` method.
   #
-  # # Usage
+  # ## Usage
   #
   # ```
   # built = LuckyMigrator::CreateTableStatement.new(:users).build do
   #   add :email : String, unique: true
   # end
   #
-  # built.table_statement
-  # #=> "CREATE TABLE users (
-  #         id serial PRIMARY KEY,
-  #         created_at timestamp NOT NULL,
-  #         updated_at timestamp NOT NULL,
-  #         email text NOT NULL);",
-  #
-  # built.index_statements
-  # #=> ["  CREATE UNIQUE INDEX users_email_index ON users USING btree (email);"]
-  #
-  # build.statements
+  # built.statements
   # # => [
   #   "CREATE TABLE users (
   #     id serial PRIMARY KEY,
   #     created_at timestamp NOT NULL,
   #     updated_at timestamp NOT NULL,
   #     email text NOT NULL);",
-  #   "  CREATE UNIQUE INDEX users_email_index ON users USING btree (email);"
+  #   "CREATE UNIQUE INDEX users_email_index ON users USING btree (email);"
   # ]
   #```
-  def build
+  def build : CreateTableStatement
     with self yield
-
-    build_table_statement
-    build_index_statements
-
     self
   end
 
-  # Join rows into table_statment and push into statements array
-  private def build_table_statement
-    statement = IO::Memory.new
-    statement << initial_table_statement
-    statement << "\n"
-
-    statement << rows.join(",\n")
-    statement << ");"
+  def statements
+    [table_statement] + index_statements
+  end
 
-    @table_statement = statement.to_s
+  private def table_statement
+    String.build do |statement|
```

Comment:
For building an string like this you can use `String.build`. It's pretty cool!

Reply:
Ah of course, I'm not sure why I used a buffer here :/